internal/ci/runner.go

```diff
@@ -97,6 +98,38 @@ func (r *Runner) setup(run config.Run) error {
 			}
 		}
 	}
+	// running pre-exec tasks
+	err := r.preExec(r.Project.Image.PreExec)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r* Runner) execute(arg string) (int, error) {
+	cmd := exec.Command(arg)
+	cmd.Stdout = r.Cli.Out()
+	cmd.Stderr = r.Cli.Out()
+	cmd.Dir = r.RunnerConfig.RootDir
+	err := cmd.Run()
+	if err != nil {
+		return 1, err
+	}
+	return 0, nil
+}
+
+func (r* Runner) preExec(preExec string) (error) {
+	tasks := utils.SplitLines(preExec)
+	for _, task := range tasks {
+		log.Info().Msg(fmt.Sprintf("Running preExec task: %s", task))
+		exitCode, err := r.execute(task)
+		if err != nil {
+			return err
+		}
+		if exitCode != 0 {
+			return fmt.Errorf("Failed to run pre-exec task: %s", task)
```

Comment:
go wants us to start error messages with a lower case 😇 
```suggestion
			return fmt.Errorf("failed to run pre-exec task: %s", task)
```